### PR TITLE
fix(react): modal now mounts child component independently of other modals

### PR DESCRIPTION
**Pull request by puopg**
Currently, when Ion components such as `IonModal` are created using the `createOverlayComponent` HOC, the isDismissing flag is scoped to the definition of the class. So in this case, when IonModal gets exported, the isDismissing flag is scoped to the IonModal class definition and not an instance of the IonModal. 

Consider when you have multiple IonModals controlling different dialogs or whatever. So I believe this `isDismissing` flag should be scoped to the instance of the Overlay class, as if I chose to render 10 IonModals, each should be controlled independently.

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: resolves https://github.com/ionic-team/ionic-framework/issues/23904

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/react/src/components/createOverlayComponent.tsx [hunk 1]
```diff
@@ -47,6 +45,7 @@ export const createOverlayComponent = <
         this.el = document.createElement('div');
       }
       this.handleDismiss = this.handleDismiss.bind(this);
+      this.isDismissing = false;
```
Comment (liamdebeasi) [inline packages/react/src/components/createOverlayComponent.tsx:48]: We need to move this out of the constructor to get the build to pass:

```typescript
class Overlay extends React.Component<Props> {
    overlay?: OverlayType;
    el!: HTMLDivElement;
    isDismissing: boolean = false;
...
```

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

File: packages/react/src/components/createOverlayComponent.tsx (hunk 1 above)
Comment (puopg) [reply packages/react/src/components/createOverlayComponent.tsx:48]: Gotcha, cool, Ill update!

**Comment by puopg** [comment]
Ready to go!

**Comment by liamdebeasi** [comment]
The code looks good, but do you have a GitHub repo that shows the issue happening? I would like to verify this fix by testing in that app.

**Comment by puopg** [comment]
@liamdebeasi Yea heres a code sandbox with ionic/react 5.7.0 showcasing the issue.

https://codesandbox.io/s/nervous-sky-w954i?file=/package.json

**Comment by puopg** [comment]
BEFORE:
======


https://user-images.githubusercontent.com/7741481/132894237-3bfc088c-79eb-46e5-b9cb-9a28e4c3bf76.mov


AFTER:
=====


https://user-images.githubusercontent.com/7741481/132894254-eb3cec1a-2d61-465d-af51-929470a6bd25.mov

**Review by liamdebeasi — APPROVED** [review]
Great job!

**Comment by liamdebeasi** [comment]
Merged. Thank you!

**Pull request merged**